frontend/static/js/views/pongView.js

```diff
@@ -2,15 +2,36 @@ import ChatSocket from '../pong/ChatSocket.js';
 import PongService from '../services/pongService.js';
 import Pong from '../pong/pong.js';
 import AView from './AView.js';
-import PongGame from '../pong/pongGame.js';
-
 
 export default class extends AView {
 	constructor(params) {
 		super(params);
 		this.setTitle('Pong');
 		this.chatSocket = null;
 		this.pongService = new PongService();
+		this.attachEventListeners = this.attachEventListeners.bind(this);
+	}
+
+	attachEventListeners() {
+		document.querySelectorAll('.nav-link').forEach((link) => {
+			link.addEventListener('click', this.chatSocket.handleClose);
+		});
+
+		window.addEventListener('beforeunload', this.chatSocket.handleClose);
+
+		const observer = new MutationObserver((mutationsList, observer) => {
```

Comment:
again the smallest thing ever but quite the nesting you got going on here :P will reverting the ifs make this a little bit more readable?

Reply:
Just noticed this comment now. I marked it as a todo for some clean up pr, I plan for the end of the week